ws/adapter.go

```diff
@@ -0,0 +1,83 @@
+package ws
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-kit/kit/log"
+	"github.com/gorilla/websocket"
+	"github.com/mainflux/mainflux"
+	broker "github.com/nats-io/go-nats"
+)
+
+var _ Service = (*adapterService)(nil)
+
+// Service contains publish and subscribe methods necessary for
+// message transfer.
+type Service interface {
+	mainflux.MessagePublisher
+
+	// HandleMessage that is received from message broker.
+	HandleMessage(*broker.Msg)
+
+	// AddConnection adds new client ws connection for given client and channel.
+	AddConnection(string, string, *websocket.Conn)
```

Comment:
Use the structure you've defined below (`idPair`), promote it, and pass it as a parameter here, instead of passing identifiers separately.